src/policy/mod.rs

```diff
@@ -25,14 +25,116 @@
 //! these with BIP32 paths, pay-to-contract instructions, etc.
 //!
 
-pub mod ast;
+mod ast;
 mod compiler;
 pub mod descriptor;
 mod embed;
 mod error;
 pub mod key;
 pub mod satisfy;
 
-pub use ast::Policy;
+use crate::miniscript::{MiniscriptKey, Translator};
+
+use std::fmt;
+
 pub use descriptor::Descriptor;
 pub use error::Error;
+
+/// Policy that expresses spending conditions for Simplicity.
+///
+/// The policy can be encoded directly as a Simplicity program which can be committed to the
+/// blockchain; when finalized, it can executed on the Bit Machine.
+///
+/// Policies can be normalized and are amenable to semantic analysis.
+#[derive(Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Policy<Pk: MiniscriptKey> {
+    fragment: ast::Fragment<Pk>,
+}
```

Comment:
04bbd20406315e83722fb7012c0e59b72fed9b51: Like @apoelstra already said, this doesn't add anything and makes the code more complex. I would remove the commit and keep it on the back burner in case we find data to cache.

Reply:
Sounds good, will do.